src/auth.py

```diff
@@ -77,9 +77,9 @@ def require_token_auth(func):
 	"""
 	@wraps(func)
 	def wrapper(*arg, **kwargs):
-		token = request.headers["Authorization"]
+		token = request.headers.get("Authorization", None)
 		course = kwargs[CID_KEY]
-		if course is None or course.get("github_token", None) != token:
+		if course is None or token is None or token != BROADWAY_API_TOKEN:
```

Comment:
Broadway is meant to be multi-course, with a roster per course. My understanding is that using the Broadway API token for authenticating the roster update would allow an admin to update the roster for any course. I think it would make more sense to use the course token, namely `course.get("token", None)`.

Reply:
When updating the code, I think I made an error using `github_token` instead of `token`.

Reply:
Let's also rename `require_token_auth` to `require_course_auth` to make the function more explicit.

Reply:
> Broadway is meant to be multi-course, with a roster per course. My understanding is that using the Broadway API token for authenticating the roster update would allow an admin to update the roster for any course. I think it would make more sense to use the course token, namely `course.get("token", None)`.

This would make sense, but I don't think course was being passed in correctly then, because kwargs[CID] would return the course id, so it's probably better to pass that in via the request.form or something.

Reply:
> When updating the code, I think I made an error using `github_token` instead of `token`.

Unless this was the issue, then it might fix it.

Reply:
the course should be received correctly, because it's a path parameter coming in through `kwargs`. I think the issue is just the token name - you and/or I can pull the code and test this route locally (would help to have unit tests haha).